cql3/result_set.hh

```diff
@@ -193,6 +194,27 @@ public:
         }
     }
 
+    // visit_gently() is like visit(), except it may yield between rows so
+    // returns a future that will resolve when it's done. We only yield
+    // between rows, not between individual cells, which is a good compromise
+    // if we assume that individual rows are not too large.
+    template<typename Visitor>
+    requires ResultVisitor<Visitor>
+    future<> visit_gently(Visitor& visitor) const {
+        auto column_count = get_metadata().column_count();
+        for (auto& row : _rows) {
+            visitor.start_row();
+            for (auto i = 0u; i < column_count; i++) {
+                auto& cell = row[i];
+                visitor.accept_value(cell ? managed_bytes_view_opt(*cell) : managed_bytes_view_opt());
+            }
+            visitor.end_row();
+            co_await coroutine::maybe_yield();
+        }
+        co_return;
```

Comment:
A coroutine here is a pessimization for the common case of just a few rows. Better use do_for_each(), which doesn't allocate anything in the common case where it doesn't need to yield.

The extra allocation is likely lost in the noise now, but let's not make things worse.

Reply:
Sure, makes sense, I'll do this.
But in general, as we move towards using coroutines more and more, often to make the code easier to read and write - I think this will be a hopeless cause. In particular, this PR changes also the caller of this function to be a coroutine. So even if this function doesn't need to allocate in case of a small result, the caller will. The question is whether you think it's worth it also in that caller to avoid using a coroutine (and have ugly indented code) just to avoid this allocation.